### PR TITLE
appplatform: upgrading to API version `2021-09-01-preview`

### PR DESCRIPTION
**Pull request by ms-henglu**
image

### CODE REVIEW

**Review by tombuildsstuff — CHANGES_REQUESTED** [review]
hi @ms-henglu

Thanks for this PR.

I've taken a look through and spotted what appears to be a breaking change with the discriminator here, as such I believe there's additional code changes required for this to work?

Thanks!

File: vendor/github.com/Azure/azure-sdk-for-go/services/preview/appplatform/mgmt/2021-09-01-preview/appplatform/models.go [hunk 1]
```diff
@@ -932,28 +994,86 @@ type CertificateProperties struct {
...
+func (cp CertificateProperties) AsBasicCertificateProperties() (BasicCertificateProperties, bool) {
+	return &cp, true
+}
+
 // CertificateResource certificate resource payload.
 type CertificateResource struct {
 	autorest.Response `json:"-"`
 	// Properties - Properties of the certificate resource payload.
-	Properties *CertificateProperties `json:"properties,omitempty"`
+	Properties BasicCertificateProperties `json:"properties,omitempty"`
```
[the remark this answers is not in the document]
Comment (ms-henglu) [reply vendor/github.com/Azure/azure-sdk-for-go/services/preview/appplatform/mgmt/2021-09-01-preview/appplatform/models.go:1076]: Yes, I also made changes to certificate resource to adopt this change.

**Comment by ms-henglu** [comment]
Hi @tombuildsstuff , would you please take another look? thanks!

**Comment by tombuildsstuff** [comment]
Rebased on top of main rather than merging main, since this is also going to conflict with #14667

**Review by tombuildsstuff — APPROVED** [review]
LGTM 👍

**Comment by tombuildsstuff** [comment]
Tests seem fine 👍

**Pull request merged**